src/linear_algebra/basic.lean

```diff
@@ -676,6 +676,9 @@ p.subtype.cod_restrict p' $ λ ⟨x, hx⟩, h hx
 
 theorem of_le_apply (h : p ≤ p') (x : p) : of_le h x = ⟨x, h x.2⟩ := rfl
 
+theorem of_le_injective (h : p ≤ p') : function.injective (of_le h) :=
+λ x y h, subtype.val_injective (subtype.mk.inj h)
```

Comment:
Does `subtype.map_injective id_injective` or similar work as a proof? Or maybe
```suggestion
subtype.val_injective.comp @@subtype.mk.inj
```
(Untested)

Reply:
I don't really see the advantage of either golf, so I'll mark this as resolved and merge.